### PR TITLE
[Bug] Modifications should be disabled when building a project (#2596)

### LINKED ISSUE #2596: [Bug] Modifications should be disabled when building a project
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### Java Version

1.8

### Scala Version

2.11.x

### StreamPark Version

2.1.0-SNAPSHOT

### Flink Version

1.3.16

### deploy mode

None

### What happened

When my project was under construction, I made some modifications to the branch and the subsequent status information changed. The last build task on the backend is currently in progress, and the modified one can be built again. The final state may conflict.

### Error Exception

```log
Status will be problematic
```

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!(您是否要贡献这个PR?)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SoberChina**
## What changes were proposed in this pull request

Issue Number: close #2596

## Verifying this change

When the build is triggered, the modify button is disabled. If an error message is displayed when modifying the project being built and saving it, errors cannot be correctly displayed when modifying the API. [code](https://github.com/apache/incubator-streampark/blob/14581ad0437b9e427b35fa661bf009f89b6e4c70/streampark-console/streampark-console-webapp/src/views/flink/project/Edit.vue#L80)

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): no

### CODE REVIEW

File: streampark-console/streampark-console-webapp/src/views/flink/project/components/ListItem.vue [hunk 1]
```diff
@@ -89,13 +89,18 @@
             </a-button>
           </a-popconfirm>
         </a-tooltip>
+        <a-tooltip :title="t('flink.project.operationTips.updateProject')">
```
Comment (wolfboys) [inline streampark-console/streampark-console-webapp/src/views/flink/project/components/ListItem.vue:92]: We need to judge whether the buildState of the current job is building, if building, disable edit button or edit button not display, else it is allowed to update

File: streampark-console/streampark-console-webapp/src/views/flink/project/components/ListItem.vue [hunk 2]
```diff
@@ -89,13 +89,18 @@
...
+          </a-button>
+        </a-tooltip>
       </template>
 
-      <a-tooltip :title="t('flink.project.operationTips.updateProject')">
-        <a-button v-auth="'project:update'" @click="handleEdit" shape="circle" class="ml-8px">
-          <EditOutlined />
-        </a-button>
-      </a-tooltip>
+<!--      <a-tooltip :title="t('flink.project.operationTips.updateProject')">-->
```
Comment (wolfboys) [inline streampark-console/streampark-console-webapp/src/views/flink/project/components/ListItem.vue]: What's this？

Comment (SoberChina) [reply streampark-console/streampark-console-webapp/src/views/flink/project/components/ListItem.vue]: Oh, sorry for not cleaning up the previous code in a timely manner.

File: streampark-console/streampark-console-webapp/src/views/flink/project/components/ListItem.vue (hunk 1 above)
Comment (SoberChina) [reply streampark-console/streampark-console-webapp/src/views/flink/project/components/ListItem.vue:92]: This logic is the same as  [building](https://github.com/apache/incubator-streampark/blob/ed33ca2c0a773222071a22cf103aa44dc5d18e29/streampark-console/streampark-console-webapp/src/views/flink/project/components/ListItem.vue#L79).

Comment (wolfboys) [reply streampark-console/streampark-console-webapp/src/views/flink/project/components/ListItem.vue:92]: > This logic is the same as [building](https://github.com/apache/incubator-streampark/blob/ed33ca2c0a773222071a22cf103aa44dc5d18e29/streampark-console/streampark-console-webapp/src/views/flink/project/components/ListItem.vue#L79).

yes, please update it.

**Review by wolfboys — APPROVED** [review]
LGTM!

**Pull request merged**